ern/graphql.go

```diff
@@ -0,0 +1,553 @@
+
+		response = append(response, &soundRecordingResolver{objectID, &soundRecording})
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+/*
+	* Release
+*/
+
+type releaseArgs struct {
+	ID   *string
+	Title   *string
+}
+
+type releaseResolver struct {
+	cid string
+	release *Release
+}
+
+func (rl *releaseResolver) Cid() string {
+	return rl.cid
+}
+
+func (rl *releaseResolver) ArtistName() string {
+	return rl.release.ArtistName
+}
+
+func (rl *releaseResolver) DisplayTitle() string {
+	return rl.release.DisplayTitle
+}
+
+func (rl *releaseResolver) Genre() *string {
+	return &rl.release.Genre
+}
+
+func (rl *releaseResolver) ReleaseType() *string {
+	return &rl.release.ReleaseType
+}
+
+func (rl *releaseResolver) ReleaseId() *string {
+	return &rl.release.ReleaseId
+}
+
+func (g *Resolver) Release(args releaseArgs) ([]*releaseResolver, error) {
+	var rows *sql.Rows
+	var err error
+
+	switch {
+	case args.ID != nil:
+		rows, err = g.db.Query("SELECT cid FROM release WHERE id = ?", *args.ID)
+	case args.Title != nil:
+		rows, err = g.db.Query("SELECT cid FROM release WHERE title = ?", *args.Title)
+	default:
+		return nil, errors.New("Missing ID or Title argument in query")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	var response []*releaseResolver
+
+	for rows.Next() {
+		var objectID string
+
+		if err := rows.Scan(&objectID); err != nil {
+			return nil, err
+		}
+
+		id, err := cid.Parse(objectID)
+
+		if err != nil {
+			return nil, err
+		}
+
+		obj, err := g.store.Get(id)
+
+		if err != nil {
+			return nil, err
+		}
+
+		var ArtistName struct {
+			Value string `json:"@value"`
+		}
+
+		if err := DecodeResolverObj(g, obj, &ArtistName, "ReleaseDetailsByTerritory", "DisplayArtist", "PartyName", "FullName"); err != nil {
+			return nil, err
+		}
+
+		// Title field can be either a single object, or multiple
+		// with varying "TitleType"s
+		// So the parent object should be loaded and then traversed accordingly
+		graph := meta.NewGraph(g.store, obj)
+		titles, err := graph.Get("ReleaseDetailsByTerritory", "Title")
+		if meta.IsPathNotFound(err) || err != nil {
+			return nil, err
+		}
+		var cids []*cid.Cid
+		switch titles := titles.(type) {
+		case *format.Link:
+			cids = []*cid.Cid{titles.Cid}
+		case []interface{}:
+			for _, x := range titles {
+				cid, ok := x.(*cid.Cid)
+				if !ok {
+					return nil, fmt.Errorf("invalid resource type %T, expected *cid.Cid", x)
+				}
+				cids = append(cids, cid)
+			}
+		}
+
+		var DisplayTitle struct {
+			Value string `json:"@value"`
+		}
+		// Load each title CID and check for the TitleType
+		for _, cid := range cids {
+			obj, err := g.store.Get(cid)
+			if err != nil {
+				return nil, err
+			}
+			tt, err := obj.Get("TitleType")
+			if err != nil {
+				return nil, err
+			}
+
+			if tt == "DisplayTitle" {
+				if err := DecodeResolverObj(g, obj, &DisplayTitle, "TitleText"); err != nil {
```

Comment:
This will overwrite the value of `DisplayTitle` on each loop iteration, is that the intention?

Reply:
Yes, because each loop is relative to a single release object and there should only be one `DisplayTitle` field on each one.